salt/_modules/azurearm_network.py

```diff
@@ -2749,3 +2749,1408 @@ def route_tables_list_all(**kwargs):
+        certificate as a Base-64 encoded string. Required only if external radius authentication has been configured
+        with EAPTLS authentication.
+
+    :param client_root_certificates: A list of client root certificates public certificate data encoded as Base-64
+        strings. Optional parameter for external radius based authentication with EAPTLS.
+ 
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call azurearm_network.virtual_network_gateway_generatevpnclientpackage test_name test_group test_params
+
+    '''
+    netconn = __utils__['azurearm.get_client']('network', **kwargs)
+
+    try:
+        pkgmodel = __utils__['azurearm.create_object_model'](
+            'network',
+            'VpnClientParameters',
+            processor_architecture=processor_architecture,
+            authentication_method=authentication_method,
+            radius_server_auth_certificate=radius_server_auth_certificate,
+            client_root_certificates=client_root_certificates
+        )
+    except TypeError as exc:
+        result = {'error': 'The object model could not be built. ({0})'.format(str(exc))}
+        return result
+
+    try:
+        pkg = netconn.virtual_network_gateways.generatevpnclientpackage(
+            resource_group_name=resource_group,
+            virtual_network_gateway_name=name,
+            parameters=pkgmodel,
+            **kwargs
+        )
+
+        result = pkg
+    except CloudError as exc:
+        __utils__['azurearm.log_cloud_error']('network', str(exc), **kwargs)
+        result = {'error': str(exc)}
+    except SerializationError as exc:
+        result = {'error': 'The object model could not be parsed. ({0})'.format(str(exc))}
+
+    return result
+
+
+def virtual_network_gateway_generate_vpn_profile(name, resource_group, processor_architecture, authentication_method,
+                                                 radius_server_auth_certificate, client_root_certificates, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Generates VPN profile for P2S client of the virtual network gateway in the 
+	specified resource group. Used for IKEV2 and radius based authentication.
+
+    :param name: The name of the virtual network gateway.
+
+    :param resource_group: The name of the resource group.
+
+    :param processor_architecture: VPN client Processor Architecture. Possible values include: 'Amd64', 'X86'
+    
+    :param authentication_method: VPN client authentication method. Possible values include: 'EAPTLS', 'EAPMSCHAPv2'
+    
+    :param radius_server_auth_certificate: The public certificate data for the radius server authentication
+        certificate as a Base-64 encoded string. Required only if external radius authentication has been configured
+        with EAPTLS authentication.
+    
+    :param client_root_certificates: A list of client root certificates public certificate data encoded as Base-64 
```

Comment:
Make radius_server_auth_certificate and client_root_certificates default to None (optional args). If they don't read the notes and don't provide it in the correct cases, then that's on them.